webapp/niapp.yaml

```diff
@@ -0,0 +1,416 @@
+      summary: Update sharing settings of the webapp
+      description: Update sharing settings of the webapp
+      operationId: update-webapp-sharing
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - $ref: '#/parameters/UpdateWebAppSharingRequest'
+      responses:
+        200:
+          $ref: '#/responses/UpdateWebAppSharingResponse'
+        400:
+          $ref: '#/responses/ValidationError'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+  '/webapps/shared-emails':
+    get:
+      tags:
+        - metadata
+      summary: List the emails the user has shared webapps with
+      description: List the emails the user has shared webapps with
+      operationId: list-shared-emails
+      responses:
+        200:
+          $ref: '#/responses/SharedEmailsResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+  '/webapps/{id}/content':
+    get:
+      tags:
+        - content
+      summary: Get webapp content
+      description: Get the content of a webapp. ContentType varies from JSON for dashboards and templates or redirect to main HTML for WebVIs
+      operationId: get-content-index
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+      responses:
+        200:
+          description: The webapp's content. JSON for Dashboards, HTML for WebVIs
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+    put:
+      tags:
+        - content
+      summary: Upload content for an existing webapp
+      description: Upload content an existing webapp
+      operationId: update-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - in: body
+          required: true
+          name: content
+          description: The webapp content to upload
+          schema:
+            $ref: '#/definitions/WebAppContent'
+      responses:
+        200:
+          description: Success
+        400:
+          $ref: '#/responses/ValidationError'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+  '/api/webapps/{id}/content/{path}':
+    get:
+      tags:
+        - content
+      summary: Get webapp content by path
+      description: Get webapp content at path. For webapps with multiple files (e.g. WebVIs)
+      operationId: get-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - $ref: '#/parameters/WebAppContentPath'
+      responses:
+        200:
+          $ref: '#/definitions/WebAppContent'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+
+  '/webapps/{id}/duplicate/{sourceId}':
+    post:
+      tags:
+        - content
+      summary: Duplicates webapp content
+      description: Duplicates webapp content
+      operationId: duplicate-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
```

Comment:
From my understanding, this should be the name of the new webapp rather than the ID.

Reply:
No, this route requires an existing webapp (metadata) as source and target

Reply:
It doesn't look like that's what is happening when I click "duplicate" from the dashboards grid in SystemLink Cloud. This is the same route, right?

If I duplicate a dashboard named "123" so that the duplicated dashboard has a name of "123-copy", I see that a POST is made to "https://www.systemlinkcloud.com/api/webapps/123-copy/duplicate/0569ab99-94c4-4ce7-af87-f99ad041763e". I don't see any other call that is creating metadata.

Reply:
@tschmittni, just wanted to make sure you saw the question I posted here.

Reply:
The /api/webapps/... is a thin wrapper over the actual webapp service API. Most of the time, it is just a pass-through but in this case the duplicate webapp route wrapper does a few extra things, e.g. duplicate the policy of the webapp, get the webapp metadata and create a new webapp before it actually calls the duplicate content route of the webapp service

I think the documentation of this route here is accurate.

Reply:
Ah, ok. I see the distinction now.